Add UDMA buffer sharing capability

matinlotfali opened the pull request:
Enables the sharing of the UDMA buffer between the DRP-AI plugin and other GStreamer elements.

This feature allows other elements in the pipeline to directly access the buffer allocated for DRP-AI processing, potentially improving efficiency when rendering on the image is not important.

More noticeably this PR has these changes:

- Adds a new `share_udma_buf` property to control this behavior.

- Introduces a new GStreamer buffer pool implementation to handle UDMA buffer allocation.

- Adds a warning message when `share_udma_buf` is enabled alongside properties that are incompatible, such as rendering framerate, time or bounding boxes.

### How to enable

Run the pipeline like below:

> gst-launch-1.0 v4l2src device=/dev/video0 **io-mode=userptr** ! drpai model=yolov3 **share-udma-buf=true** ! fakesink

matinlotfali commented:
Sadly stopped working :cry:

mistyron commented:
> Sadly stopped working 😢

Is there any follow up on this?

matinlotfali commented:
Should prioritize other stuff. Will get back on it later.

mistyron reviewed (changes requested):
review: TBD (copied from the description of the PR) :)

matinlotfali committed:
Add UDMA buffer sharing capability

Enables the sharing of the UDMA buffer between the DRP-AI plugin and other GStreamer elements.

This feature allows other elements in the pipeline to directly access the buffer allocated for DRP-AI processing, potentially improving efficiency.

Adds a new `share_udma_buf` property to control this behavior.

Introduces a new GStreamer buffer pool implementation to handle UDMA buffer allocation.

Adds a warning message when `share_udma_buf` is enabled alongside properties that are incompatible, such as rendering framerate, time or bounding boxes.

matinlotfali commented:
It is now working. Updated the descriptions

kieferMaland reviewed (approved):
Reviewed at a high level and tested on SOM with and without showing fps and bboxes. 
LGTM.
Great work!

matinlotfali committed:
Merge branch 'develop' into share-udma-buf
